### DISCUSSION TITLE
Standard Naming Convention for Copilot Instruction Files

### DISCUSSION THREAD

**Discussion by fielding** [fielding/copilot-instructions#1] [General]
# Standard Naming Convention for Copilot Instruction Files

As we continue to develop and refine this repository, one of the key considerations is establishing a **standard naming convention** for the Copilot instruction files. Currently, we're using `.vscode/copilot-instructions.md` as the standard file name and location.

**Purpose of This Discussion:**

- **Gather Feedback:** We want to hear your thoughts on this naming convention.
- **Explore Alternatives:** Are there other naming conventions or file locations that might be more effective?
- **Establish a Community Standard:** By collaborating, we can agree on a standard that benefits everyone.

**Points to Consider:**

- **Consistency with Other Tools:** Does the naming convention align with common practices in other tools or extensions?
- **Ease of Use:** Is the file name and location intuitive for users of different experience levels?
- **Compatibility:** Are there any issues with how GitHub Copilot reads or processes instruction files in certain locations?
- **Ease of Discoverability:** This is my main personal motivation. It's fun to see what others are doing sometimes.

**Questions for the Community:**

1. **Do you prefer `.vscode/copilot-instructions.md`, or is there a better alternative?**
2. **Should the instruction file be placed in a different directory, such as the project's root directory?**
3. **Have you encountered any issues with the current naming convention?**

**How to Participate:**

- Share your thoughts, experiences, and suggestions by replying to this discussion.
- If you have alternative proposals, please explain the reasoning behind them.

**Comment by katriendg** [comment]
Thanks for starting this discussion. Would love to hear more input from folks following or preferring certain approaches.

What I'm seeing is we use many different contexts setting `.md` files which we have not yet standardized on:

**Locations**

- Do we put the context/instruction file within the application folder (like `deploy` `/src/language/projectname`, etc. The latter leading to polluting the folder structure with copilot context information.
- Do we standardize in a central location such as `.vscode/copilot` or `.github/copilot`, `.github/`...

**File name conventions**

- `.copilot-someinstruction.md`
- `copilot-someinstruction.md` - no use of initial `.`
- `copilot.someinstruction.md`
- Copilot template files which we use to build a documentation of our feature, and then use that documentation to generate a specific piece of code implementation: `/templates/myfunction.md` which is added to Copilot context.

I'd be really interested in learning what folks are finding most efficient. We are still experimenting with these above different approaches.
